### PR TITLE
More shiny python

### PR DESCRIPTION
**Pull request by wesleyw72**
Added some Python shiny docs/examples for:
 - Autocorrelation
 - Thermometers example
 - Bit about using ndarrays in Python instead of tensors

### CODE REVIEW

File: docs/doc_templates/autocorrelation.md [hunk 1]
```diff
@@ -41,3 +43,18 @@ For example, if the sample shape is `[1,5]` we can evaluate the autocorrelation
...
+
+#### Python
+
+It's also possible to calculate the autocorrelation of samples in Python.
+
+```python
+{% snippet PythonScalarAutocorrelation %}
+```
+
+If sampling `ndarray`'s in Python. The index on which to calculate the autocorrelation can be specified
```
Comment (migwellian) [inline docs/doc_templates/autocorrelation.md]:
```suggestion
If sampling `ndarray`s in Python, the index on which to calculate the autocorrelation can be specified 
```

File: docs/doc_templates/tensors.md [hunk 2]
```diff
@@ -148,3 +148,14 @@ In Keanu the above problem would be written as:
 ```java
 {% snippet TensorFinal %}
 ```
+
+## Python
+
+We do not expose Tensors in the Python API. Numpy provides the same concept through its `ndarray` class. 
+
+Most of the operations that used Tensor within Java can be replaced with `ndarray` in Python.
+For example, to create a vertex.
```
Comment (migwellian) [inline docs/doc_templates/tensors.md]:
```suggestion
For example, to create a vertex:
```

File: docs/current_docs/tensors.md [hunk 3]
```diff
@@ -177,9 +177,23 @@ DoubleVertex CObservation = new GaussianVertex(C, 1);
...
+Optimizer optimizer = KeanuOptimizer.of(new BayesianNetwork(A.getConnectedGraph()));
 optimizer.maxAPosteriori();
 
 //Retrieve the most likely estimate using MAP estimation
 DoubleTensor mostLikelyEstimate = A.getValue(); //approximately [2, 3]
 ```
+
+## Python
+
+We do not expose Tensors in the Python API. Numpy provides the same concept through its `ndarray` class.
```
Comment (GeorgeNash) [inline docs/current_docs/tensors.md]: through its `ndarray` class and these are converted to Tensors for you.

File: docs/current_docs/tensors.md [hunk 4]
```diff
@@ -177,9 +177,23 @@ DoubleVertex CObservation = new GaussianVertex(C, 1);
...
 
 //Retrieve the most likely estimate using MAP estimation
 DoubleTensor mostLikelyEstimate = A.getValue(); //approximately [2, 3]
 ```
+
+## Python
+
+We do not expose Tensors in the Python API. Numpy provides the same concept through its `ndarray` class. 
+
+Most of the operations that used Tensor within Java can be replaced with `ndarray` in Python.
```
Comment (GeorgeNash) [inline docs/current_docs/tensors.md]: `All of the operations` (hopefully)?

File: docs/current_docs/tensors.md [hunk 5]
```diff
@@ -177,9 +177,23 @@ DoubleVertex CObservation = new GaussianVertex(C, 1);
...
+For example, to create a vertex.
+
+```python
+mu = np.array([[2., 3., 4.],
+              [1., 4., 7.]])
+sigma = np.ones([2,3])
```
Comment (GeorgeNash) [inline docs/current_docs/tensors.md]: nit: `[2, 3]`

File: docs/current_docs/autocorrelation.md [hunk 6]
```diff
@@ -51,3 +53,24 @@ NetworkSamples posteriorSamples = MetropolisHastings.withDefaultConfig().getPost
...
+It's also possible to calculate the autocorrelation of samples in Python.
+
+```python
+posterior_samples = sample(net=bayes_net, sample_from=bayes_net.get_latent_vertices(),
+                           algo="metropolis", draws=100)
+vertex_samples = posterior_samples.get('a')
+ac = stats.autocorrelation(vertex_samples)
+```
+
+If sampling `ndarray`'s in Python. The index on which to calculate the autocorrelation can be specified
```
Comment (GeorgeNash) [inline docs/current_docs/autocorrelation.md]: When the samples are `ndarray's` the index...

File: docs/current_docs/tensors.md (hunk 3 above)
Comment (wesleyw72) [reply docs/current_docs/tensors.md]: Do we need to mention this though? It's completely hidden from the user.

Comment (GeorgeNash) [reply docs/current_docs/tensors.md]: Aye, as people will be familiar with Tensors so it's good to ensure they know the translation is happening behind the scenes. `ndarray` is just another face on top of them

**Pull request merged**